id_minter/src/test/scala/uk/ac/wellcome/platform/idminter/steps/IdentifierGeneratorTest.scala

```diff
@@ -38,43 +40,31 @@ class IdentifierGeneratorTest
 
   it("should generate an id and save it in the database if a record doesn't already exist") {
     val work =
-      Work(
-        identifiers = List(SourceIdentifier("Miro", "MiroID", "1234")), label = "some label")
+      Work(identifiers = List(SourceIdentifier("Miro", "MiroID", "1234")),
+           label = "some label")
     val futureId = identifierGenerator.generateId(work)
 
     whenReady(futureId) { id =>
       id should not be (empty)
-      Scanamo.queryIndex[Identifier](dynamoDbClient)(identifiersTableName,
-                                                     "MiroID")(
-        'MiroID -> "1234") shouldBe List(Right(Identifier(id, "1234")))
+
+      val i = Identifiers.syntax("i")
+      val maybeIdentifier = withSQL {
+        select.from(Identifiers as i).where.eq(i.MiroID, "1234")
+      }.map(Identifiers(i)).single.apply()
+      maybeIdentifier shouldBe defined
+      maybeIdentifier.get shouldBe Identifier(id, "1234")
     }
   }
 
   it("should reject an item with no miroId in the list of Identifiers") {
     val work =
       Work(
-        identifiers = List(SourceIdentifier("NotMiro", "NotMiroID", "1234")), label = "some label")
+        identifiers = List(SourceIdentifier("NotMiro", "NotMiroID", "1234")),
+        label = "some label")
     val futureId = identifierGenerator.generateId(work)
 
     whenReady(futureId.failed) { exception =>
       exception.getMessage shouldBe s"Item $work did not contain a MiroID"
     }
   }
-
-  it("should return an error if it finds more than one record for the same MiroID") {
```

Comment:
It's impossible with Mysql, isn't it? We should add another one where we test that it throws an exception if we try to add a record for an already existing miroID